Solutions/Marain.UserNotifications.Specs/Features/ManagementApiClient/NotificationTemplate.feature

```diff
@@ -0,0 +1,44 @@
+﻿@perFeatureContainer
+@useApis
+@useTransientTenant
+
+Feature: Notification Template via the client library
+
+Scenario: Create a Notification Template
+	When I use the client to send the notification template API a request to create a new notification template
+		"""
+		{
+			"notificationType": "Marain.Notification.NewLead.v1",
+			"smsTemplate": 
+			{
+				"body": "A new lead was added by {{leadAddedBy}}"
+			}
+		}
+		"""
+	Then the client response status code should be 'OK'
+
+Scenario: Update a Notification Template
+	Given I have created and stored a notification template
+	| notificationType               | smsTemplate                                         |
+	| Marain.Notification.NewLead.v1 | {"body": "A new lead was added by {{leadAddedBy}}"} |
+	When I use the client to send the notification template API a request to create a new notification template
+		"""
+		{
+			"notificationType": "Marain.Notification.NewLead.v1",
+			"smsTemplate": 
+			{
+				"body": "Different template"
+			}
+		}
+		"""
+	Then the client response status code should be 'OK'
+	# And the response content should have a json property called 'sms' with value '{"body": "Different template"}'
```

Comment:
You aren't actually validating that it did anything. If you get it, having updated it, are you expecting the new one immediately, or is there some cache expiry; what about ETags? If I get a template, and update it, and someone else updates it, are we just "latest wins"?

Thinking more broadly: is the code safe for multiple simultaneous creators/updaters? What if I was doing a create and someone got in with a create before me? What errors/successes should I expect? Shouldn't we be supporting ETags so we get sensible concurrency errors?

Reply:
We have added eTags support to handle the concurrency issues.